### PR TITLE
Add big object support for aarch64 PE

### PR DESCRIPTION
**Pull request by vejbomar**
Adds support for big objects in the aarch64 COFF target in `aarch64-w64-mingw32-as` which can be enabled with `-mbig-obj` argument. Without this creating binary with a lot of sections fails with the error like:

```
/home/vejby/cross-aarch64-w64-mingw32-msvcrt/bin/aarch64-w64-mingw32-as: a.out: too many sections (50004)
code/binutils/gas/testsuite/gas/pe/big-obj.s: Assembler messages:
code/binutils/gas/testsuite/gas/pe/big-obj.s: Fatal error: can't write 1 byte to section .data$a00000 of a.out: 'file too big'
/home/vejby/cross-aarch64-w64-mingw32-msvcrt/bin/aarch64-w64-mingw32-as: a.out: too many sections (50004)
code/binutils/gas/testsuite/gas/pe/big-obj.s: Fatal error: a.out: file too big
```

Based on [this MSVC documentation](https://learn.microsoft.com/en-us/cpp/build/reference/bigobj-increase-number-of-sections-in-dot-obj-file?view=msvc-170), this should manifest if you want to create more than 2^16 sections but it actually happens already somewhere between 2^15 and 2^16.

### CODE REVIEW

**Comment by eukarpov** [comment]
LGTM. CI should be extended with a big-obj test and triggered to validate current change.
https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build

**Comment by vejbomar** [comment]
> LGTM. CI should be extended with a big-obj test and triggered to validate current change. https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build

Afaik we don't run bintuils tests in https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build, so do you suggest to implement something like [big-obj.s](https://github.com/Windows-on-ARM-Experiments/binutils-woarm64/blob/woarm64/gas/testsuite/gas/pe/big-obj.s) and [big-obj.d](https://github.com/Windows-on-ARM-Experiments/binutils-woarm64/blob/woarm64/gas/testsuite/gas/pe/big-obj.d) in https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/tree/main/tests ?

**Comment by eukarpov** [comment]
Yes, some script which will try to compile big-obj.s and validate the result with and without the flag.

File: bfd/pe-aarch64.c [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
 #define TARGET_SYM             aarch64_pe_le_vec
 #define TARGET_NAME            "pe-aarch64-little"
+#define TARGET_SYM_BIG		     aarch64_pe_big_vec
```
Comment (Blackhex) [inline bfd/pe-aarch64.c]: Inconsistent alignment.

File: bfd/targets.c [hunk 2]
```diff
@@ -996,6 +997,7 @@ static const bfd_target * const _bfd_target_vector[] =
 	&aarch64_elf64_le_cloudabi_vec,
 	&aarch64_mach_o_vec,
 	&aarch64_pe_le_vec,
+    &aarch64_pe_big_vec,
```
Comment (Blackhex) [inline bfd/targets.c]: Inconsistent indentation.

File: gas/config/tc-aarch64.c [hunk 3]
```diff
@@ -11059,6 +11068,12 @@ md_parse_option (int c, const char *arg)
       break;
 #endif
 
+# if defined (TE_PE) || defined (TE_PEP)
```
Comment (Blackhex) [inline gas/config/tc-aarch64.c:11071]: \*\`#if\`

File: gas/config/tc-aarch64.c [hunk 4]
```diff
@@ -10137,7 +10142,7 @@ aarch64elf_frob_symbol (symbolS * symp, int *puntp)
 const char *
 coff_aarch64_target_format (void)
 {
-  return "pe-aarch64-little";
+  return use_big_obj ? "pe-aarch64-bigobj" : "pe-aarch64-little";
```
Comment (Blackhex) [inline gas/config/tc-aarch64.c:10145]: This usage of `use_big_obj` is not guarded by `#if defined (TE_PE) || defined (TE_PEP)`. Wouldn't it lead to compilation error?

**Review by Blackhex — CHANGES_REQUESTED** [review]
Except of https://github.com/Windows-on-ARM-Experiments/binutils-woarm64/pull/5/files#r1848293155 concern, LGTM.

File: gas/config/tc-aarch64.c (hunk 4 above)
Comment (eukarpov) [reply gas/config/tc-aarch64.c:10145]: It is guarded by OBJ_COFF and likely it works. Perhaps it makes sense to use OBJ_COFF for use_big_obj declaration.

**Comment by vejbomar** [comment]
> Yes, some script which will try to compile big-obj.s and validate the result with and without the flag.

Test implemented in https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/pull/227

File: gas/config/tc-aarch64.c (hunk 4 above)
Comment (vejbomar) [reply gas/config/tc-aarch64.c:10145]: Changed all guarding to `OBJ_COFF` which seems to be used more in the `tc-aarch64.c`

Implementation tested here: https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/runs/12292436394

**Review by Blackhex — APPROVED** [review]
LGTM now.

**Review by eukarpov — APPROVED** [review]
CI validation
https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/runs/12292436394

**Pull request merged**